--- origsrc/giggle-0.4.91/libgiggle/Makefile.am	2009-02-14 17:28:25.000000000 -0600
+++ src/giggle-0.4.91/libgiggle/Makefile.am	2009-06-21 20:37:32.896698900 -0500
@@ -62,6 +62,7 @@ libgiggle_la_LIBADD = \
 libgiggle_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
 	-release $(PACKAGE_VERSION) \
+	-no-undefined \
 	-Wl,--as-needed -Wl,--no-undefined \
 	$(NULL)
 
--- origsrc/giggle-0.4.91/libgiggle-git/Makefile.am	2009-02-14 17:28:25.000000000 -0600
+++ src/giggle-0.4.91/libgiggle-git/Makefile.am	2009-06-21 20:37:20.281708400 -0500
@@ -67,6 +67,7 @@ libgiggle_git_la_LIBADD = \
 libgiggle_git_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
 	-release $(PACKAGE_VERSION) \
+	-no-undefined \
 	-Wl,--as-needed -Wl,--no-undefined \
 	$(NULL)
 
--- origsrc/giggle-0.4.91/plugins/Makefile.am	2009-02-12 15:34:07.000000000 -0600
+++ src/giggle-0.4.91/plugins/Makefile.am	2009-06-21 21:42:34.630957100 -0500
@@ -7,7 +7,8 @@ AM_CPPFLAGS = \
 
 AM_LDFLAGS = \
 	$(PLATFORM_LDFLAGS) \
-	-module -rpath "$(pkglibdir)/plugins"
+	-module -avoid-version -no-undefined \
+	-rpath "$(pkglibdir)/plugins"
 
 plugindir = $(pkglibdir)/plugins
 
@@ -24,10 +25,12 @@ dist_noinst_DATA = \
 libhello_world_la_SOURCES = \
 	giggle-hello-world-plugin.c
 
+libhello_world_la_LIBADD = $(top_builddir)/libgiggle/libgiggle.la $(GIGGLE_LIBS)
+
 libpersonal_details_la_SOURCES = \
 	giggle-personal-details-plugin.c \
 	giggle-personal-details-window.c \
 	giggle-personal-details-window.h
 
 libpersonal_details_la_CFLAGS = $(EBOOK_CFLAGS)
-libpersonal_details_la_LIBADD = $(EBOOK_LIBS)
+libpersonal_details_la_LIBADD = $(top_builddir)/libgiggle-git/libgiggle-git.la $(GIGGLE_LIBS) $(EBOOK_LIBS)
